src/test/java/org/influxdb/dto/BoundParameterQueryTest.java

```diff
@@ -0,0 +1,188 @@
+package org.influxdb.dto;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.influxdb.dto.BoundParameterQuery.QueryBuilder;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import com.squareup.moshi.JsonReader;
+
+import okio.Buffer;
+
+/**
+ * Test for the BoundParameterQuery DTO.
+ */
+@RunWith(JUnitPlatform.class)
+public class BoundParameterQueryTest {
+
+  @Test
+  public void testSingleCharacterPlaceHolderParsing() throws IOException {
+    BoundParameterQuery query = QueryBuilder.newQuery("SELECT * FROM abc WHERE a > $a AND b < $b")
+        .forDatabase("foobar")
+        .bind("a", 0)
+        .bind("b", 10)
+        .create();
+    Map<String, Object> params = readObject(decode(query.getParameterJsonWithUrlEncoded()));
+    Assert.assertEquals(2, params.size());
```

Comment:
I really appreciate your detailed and extensive test but I think it's too much: you are testing multiple times the same logic over and over.

Could you please test `query.getParameterJsonWithUrlEncoded()` only once, covering all data types supported by InfluxDB?

Reply:
Relevant code has been removed